src/main/scala/io/gatling/build/compile/GatlingCompilerSettingsPlugin.scala

```diff
@@ -25,44 +25,68 @@ object GatlingCompilerSettingsPlugin extends AutoPlugin {
 
   override def trigger = allRequirements
 
+  trait GatlingCompilerSettingsKeys {
+    val javaReleaseVersion = settingKey[String]("Java release releaseVersion")
+  }
+
+  object GatlingCompilerSettingsKeys extends GatlingCompilerSettingsKeys
+  object autoImport extends GatlingCompilerSettingsKeys
+
+  import autoImport._
+
+  private val DefaultJavaReleaseVersion = "1.8"
   private val isJava8 = scala.util.Properties.javaVersion.startsWith("1.8")
 
-  private val JavacOptions =
-    if (isJava8) {
-      Seq("-source", "1.8", "-target", "1.8")
+  private val JavacOptions = Def.setting {
+    val version = javaReleaseVersion.value
+    if (version == DefaultJavaReleaseVersion) {
+      if (isJava8) {
+        Seq("-source", "1.8", "-target", "1.8")
+      } else {
+        Seq("--release", "1.8")
+      }
     } else {
-      Seq("--release", "8")
+      // When user overwrite java release version, we leave him the responsibility
```

Comment:
As javacOptions are defined only with that. I won't bother to have a special case for that.
Plugin users (module) can already override such configuration.